src/Microsoft.DotNet.CodeFormatting/Rules/MarkReadonlyFieldsRule.cs

```diff
@@ -0,0 +1,527 @@
+                        return IsSymbolVisibleOutsideSolution(
+                            symbol.ContainingType,
+                            internalsVisibleToAttribute,
+                            solution);
+                    }
+
+                    // They are public, we are going to skip them.
+                    return true;
+                }
+
+                if (accessibility > Accessibility.Private)
+                {
+                    bool visibleOutsideSolution = IsVisibleOutsideSolution(
+                        symbol,
+                        internalsVisibleToAttribute,
+                        solution);
+
+                    if (visibleOutsideSolution)
+                    {
+                        if (symbol.ContainingType != null)
+                        {
+                            // a visible symbol in a non-visible class isn't visible
+                            return IsSymbolVisibleOutsideSolution(
+                                symbol.ContainingType,
+                                internalsVisibleToAttribute,
+                                solution);
+                        }
+
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+
+            private bool IsVisibleOutsideSolution(
+                ISymbol field,
+                ISymbol internalsVisibleToAttribute,
+                Solution solution)
+            {
+                bool isVisible;
+                IAssemblySymbol assembly = field.ContainingAssembly;
+                if (_visibleOutsideAsembly.TryGetValue(assembly, out isVisible))
+                {
+                    return isVisible;
+                }
+
+                foreach (AttributeData internalsVisibleInstance in assembly.GetAttributes()
+                    .Where(a => Equals(a.AttributeClass, internalsVisibleToAttribute)))
+                {
+                    if (internalsVisibleInstance.ConstructorArguments.Length != 1)
+                    {
+                        // Unexpected number of agruments, isn't really the correct type.
+                        continue;
+                    }
+
+                    TypedConstant assemblyNameArgument = internalsVisibleInstance.ConstructorArguments[0];
+                    if (assemblyNameArgument.Kind != TypedConstantKind.Primitive)
+                    {
+                        // The first argument wasn't a primitave value, isn't really the correct type
+                        continue;
+                    }
+
+                    string assemblyName = assemblyNameArgument.Value as string;
+                    if (String.IsNullOrEmpty(assemblyName))
+                    {
+                        // The first argument wasn't a string, isn't really the correct type
+                        continue;
+                    }
+
+                    if (!solution.Projects.Any(p => ProjectHasName(p, assemblyName)))
```

Comment:
I worry that anything that has unit tests is going to expose their internals to the test assembly, which would disable readonly for any of the internal fields.  Running on the solution that contained both of them would solve the problem.  Running on the solution will mark more things (correctly) as readonly than running it only on the project.

However, it looks like most of the framework doesn't actually expose many of it's internals, so I'll remove the solution level check for now.  I'll still have the commit in my fork if we want to add it back later.